### PR TITLE
London Class 7 - Laura M - JavaScript Core 1 - Week 1

### PR DESCRIPTION
**Pull request by Laura-TM**
# Your Details

- Your Name: Laura M
- Your City: London
- Your Slack Name: Laura M

# Homework Details

- Module: Javascript Core 1
- Week: 1

### CODE REVIEW

File: exercises/F-strings-methods/exercise.js [hunk 1]
```diff
@@ -6,8 +6,13 @@
...
 function getGreeting(name) {
-    var myGreeting = 'My name is ' + name + ' and my name is ' + name.length + ' characters long';
-    return myGreeting;
+  var myGreeting =
+    "My name is " +
+    name +
+    " and my name is " +
+    name.length +
+    " characters long";
+  return myGreeting;
```
Comment (omarsharif-gr) [inline exercises/F-strings-methods/exercise.js:15]: Great Job! As a challenge, you can make use of template literals to tidy up the code and make it easier to read. This link: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals and video: https://youtu.be/NgF9-pdTDGs are quite useful - should you need them.

Comment (Laura-TM) [reply exercises/F-strings-methods/exercise.js:15]: Thank you! It auto-formatted that way on save :-)

File: mandatory/3-function-output.js [hunk 2]
```diff
@@ -1,18 +1,30 @@
...
   return w1.concat(w2);
 }
 
 function concatenate(firstWord, secondWord, thirdWord) {
   // Write the body of this function to concatenate three words together.
   // Look at the test case below to understand what this function is expected to return.
+  let myFirstWord = firstWord + " ";
+  let mySecondWord = secondWord + " ";
+  let myThirdWord = thirdWord;
+  return myFirstWord.concat(mySecondWord, myThirdWord);
```
Comment (amellie) [inline mandatory/3-function-output.js:21]: Good effort on this Laura. Can you please think of a way to change this solution into a one-liner that is easy to read? Hint: use one only 1 method of concatenating strings. Let me know if you need further details on this

File: mandatory/3-function-output.js [hunk 3]
```diff
@@ -1,18 +1,30 @@
...
+// This 'getNumber' function will return a random number between 0 (inclusive) and 10 (exclusive), which cannot be chosen or reset by the user.
 function getNumber() {
   return Math.random() * 10;
 }
 
 // Add comments to explain what this function does. You're meant to use Google!
+
+// This 's' function with the .concat() method is used to merge two or more arrays, by returning a new array and not changing the existing arrays.
 function s(w1, w2) {
   return w1.concat(w2);
```
Comment (amellie) [inline mandatory/3-function-output.js:12]: Are you sure the ``concat`` method is used to merge arrays?

File: mandatory/4-tax.js [hunk 4]
```diff
@@ -1,11 +1,18 @@
...
+/* 
   SALES TAX
   =========
   A business requires a program that calculates how much sales tax to charge
   Sales tax is 20% of the price of the product
 */
 
-function calculateSalesTax() {}
+// This function aims to calculate the tax amount to charge on any given product depending on its price and the 20% given sales tax
+function calculateSalesTax(priceOfProduct) {
```
Comment (amellie) [inline mandatory/4-tax.js:9]: It looks like this file contains an error and the test cases failed. Would you be able to fix this?

**Review by amellie — CHANGES_REQUESTED** [review]
Well done on your work, Laura! Just a few errors in the code. Would you be able to fix them please?

File: mandatory/3-function-output.js (hunk 3 above)
Comment (Laura-TM) [reply mandatory/3-function-output.js:12]: sorry, bad use of terminology; it has now been changed :-)

File: mandatory/3-function-output.js (hunk 2 above)
Comment (Laura-TM) [reply mandatory/3-function-output.js:21]: thank you for the suggestion

File: mandatory/4-tax.js (hunk 4 above)
Comment (Laura-TM) [reply mandatory/4-tax.js:8]: there was an error on the tests, which were raised with CYF and they have now been amended :-)

**Pull request closed without merging**